File: crashfix/src/main/java/com/janbean/crashfix/NamedThreadFactory.java
package com.janbean.crashfix;

import java.util.concurrent.Executors;
import java.util.concurrent.ThreadFactory;
import java.util.concurrent.atomic.AtomicInteger;

/**
 * Created by huangjianbin on 2022/5/20
 */

public class NamedThreadFactory implements ThreadFactory {
    private final ThreadFactory mDefaultThreadFactory = Executors.defaultThreadFactory();
    private final String mBaseName;
    private final AtomicInteger mCount = new AtomicInteger(0);
    private final int mPriority;

    public NamedThreadFactory(String baseName, int priority) {
        this.mBaseName = baseName;
        this.mPriority = priority;
    }

    public Thread newThread(Runnable runnable) {
        Thread thread = this.mDefaultThreadFactory.newThread(runnable);
        thread.setName(this.mBaseName + "-" + this.mCount.getAndIncrement());
        thread.setPriority(this.mPriority);
        return thread;
    }
}

File: optimize-firebase/src/main/java/com/janbean/optimize/firebase/NamedThreadFactory.kt
package com.janbean.optimize.firebase

import android.os.Process
import java.util.concurrent.Executors
import java.util.concurrent.ThreadFactory

class NamedThreadFactory(private val name: String) : ThreadFactory {
    private val threadFactory = Executors.defaultThreadFactory()
    override fun newThread(var1: Runnable): Thread {
        val runnable = WrapRunnable(var1, 0)
        val thread = threadFactory.newThread(runnable)
        thread.name = name
        return thread
    }
}


internal class WrapRunnable(private val zza: Runnable, val priority: Int) : Runnable {
    override fun run() {
        Process.setThreadPriority(priority)
        zza.run()
    }
}

